src/main/java/nl/tudelft/jpacman/points/PointCalculatorLoader.java

```
@@ -1,19 +1,54 @@
 package nl.tudelft.jpacman.points;
 
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Properties;
+
 /**
- * The responsibility of this loader is to obtain
- * the appropriate points calculator.
- * By default the Default calculator is returned.
+ * The responsibility of this loader is to obtain the appropriate points calculator.
+ * By default the {@link DefaultPointCalculator} is returned.
  */
 public class PointCalculatorLoader {
 
+    private static Class clazz = null;
+
     /**
      * Load a points calculator and return it.
      *
      * @return The (dynamically loaded) points calculator.
      */
     public PointCalculator load() {
-        // TODO: Azqa, dynamic load here
-        return new DefaultPointCalculator();
+        try {
+            if (clazz == null) {
+                clazz = loadClassFromFile();
+            }
+
+            return (PointCalculator) clazz.newInstance();
+        } catch (Exception e) {
+            throw new RuntimeException("Could not dynamically load the points calculator.", e);
```

Yes, thanks, this is better. The exception swallowing was painful to my eyes.